search.go

```diff
@@ -0,0 +1,30 @@
+package filter
+
+import (
+	"gorm.io/gorm"
+	"goyave.dev/goyave/v4/helper"
+	"strings"
+)
+
+type Search struct {
+	Field string
+	Query string
+}
+
+// Scope returns the GORM scope to use in order to apply sorting.
+func (s *Search) Scope(settings *Settings, modelIdentity *modelIdentity) func(*gorm.DB) *gorm.DB {
+	if !helper.ContainsStr(settings.FieldsSearch, s.Field) {
+		return nil
+	}
+
+	_, ok := modelIdentity.Columns[s.Field]
+	if !ok {
+		return nil
+	}
+
+	return func(tx *gorm.DB) *gorm.DB {
+		tableWithColumn := []string{tx.Statement.Quote(modelIdentity.TableName), tx.Statement.Quote(s.Field)}
+
+		return tx.Or("(lower(CAST("+ strings.Join(tableWithColumn, ".") + " AS TEXT))) LIKE lower(?)", "%" + helper.EscapeLike(s.Query) + "%")
```

Comment:
To avoid conflict with filters, I think it would be better to use a [grouped condition](https://gorm.io/docs/advanced_query.html#Group-Conditions).

That also means that search would generate only one scope instead of n scopes (n being the number of `FieldsSearch`).

Reply:
I think the feature would be more flexible if it used an `Operator` instead of hardcoded condition (that may not work with all database engines). The `Operator` could be specified by the client through the query param.